src/cryptography/hazmat/backends/openssl/ciphers.py

```diff
@@ -107,7 +109,7 @@ def update(self, data):
         # should be taken only when length is zero and mode is not GCM because
         # AES GCM can return improper tag values if you don't call update
         # with empty plaintext when authenticating AAD for ...reasons.
-        if len(data) == 0 and not isinstance(self._mode, modes.GCM):
+        if len(data) == 0 and not self._is_gcm:
```

Comment:
This isn't actually called every `update` unless the data length is 0, so that's part of why you're not seeing much performance gain here.